src/meta-srv/src/election/postgres.rs

```diff
@@ -0,0 +1,569 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::time::{self, Duration};
+
+use common_meta::distributed_time_constants::{META_KEEP_ALIVE_INTERVAL_SECS, META_LEASE_SECS};
+use common_meta::kv_backend::postgres::{
+    CAS, POINT_DELETE, POINT_GET, PREFIX_SCAN, PUT_IF_NOT_EXISTS,
+};
+use common_telemetry::{error, info, warn};
+use serde::{Deserialize, Serialize};
+use snafu::{ensure, ResultExt};
+use tokio::sync::broadcast;
+use tokio::sync::broadcast::error::RecvError;
+use tokio::time::{timeout, MissedTickBehavior};
+use tokio_postgres::Client;
+
+use crate::election::{
+    Election, LeaderChangeMessage, LeaderKey, CANDIDATES_ROOT, CANDIDATE_LEASE_SECS, ELECTION_KEY,
+    KEEP_ALIVE_INTERVAL_SECS,
+};
+use crate::error::{
+    DeserializeFromJsonSnafu, NoLeaderSnafu, PostgresExecutionSnafu, Result, SerializeToJsonSnafu,
+    UnexpectedSnafu,
+};
+use crate::metasrv::{ElectionRef, LeaderValue, MetasrvNodeInfo};
+
+const CAMPAIGN: &str = "SELECT pg_try_advisory_lock(1)";
```

Comment:
We might need to make this keyid configurable in case that it's in use by other applications. But for now we can just leave a comment here.

Reply:
Sure. I'd like to make key-id and the table name of pg backend (currently it's hard-coded as greptime_metakv) configurable in the next pr.